tests/virus_total_agent_test.py

```diff
@@ -176,6 +176,151 @@ def testVirusTotalAgent_whenVirusTotalApiReturnsValidResponse_noExceptionRaised(
     assert agent_mock[0].data["references"] == [
         {"title": "Virustotal", "url": "https://www.virustotal.com/"}
     ]
+    assert (
+        agent_mock[0].data["dna"]
+        == '{"location": {"file": {"content": null, "content_url": null, "path": "some/dummy/path"}, "metadata": [{"type": "FILE_PATH", "value": "some/dummy/path"}]}, "title": "VirusTotal scan flagged malicious asset(s) (MD5 based search)"}'
+    )
+
+
+def testVirusTotalAgent_whenApkMessage_noExceptionRaised(
+    mocker: plugin.MockerFixture,
+    agent_mock: list[msg.Message],
+    virustotal_agent: virus_total_agent.VirusTotalAgent,
+    apk_message: msg.Message,
+) -> None:
+    """Unittest for the lifecycle of the virustotal agent :
+    Sends a dummy malicious file through the Virus Total public API,
+    receives a valid response, assign a risk rating, creates a technical detail
+    and finally emits a message of type v3.report.vulnerability with the details above.
+    """
+
+    mocker.patch(
+        "virus_total_apis.PublicApi.get_file_report",
+        side_effect=virustotal_valid_response,
+    )
+    virustotal_agent.process(apk_message)
+
+    assert len(agent_mock) == 1
+    assert all(msg.selector == "v3.report.vulnerability" for msg in agent_mock)
```

Comment:
```suggestion
    assert all(msg.selector == "v3.report.vulnerability" for msg in agent_mock) is True
```

Apply to the rest please.

Reply:
Updated